### PR TITLE
Island: Disable session-based cookies in Flask

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Fixes part of #2157.

## PR Checklist
* [ ] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?
* [ ] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] Added relevant unit tests?
* [ ] Have you successfully tested your changes locally? Elaborate:
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [ ] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: monkey/monkey_island/cc/app.py [hunk 1]
```diff
@@ -88,6 +89,7 @@ def setup_authentication(app, data_dir):
     # Ignore CSRF, because it's irrelevant for javascript applications
     app.config["WTF_CSRF_CHECK_DEFAULT"] = False
     app.config["SECURITY_CSRF_IGNORE_UNAUTH_ENDPOINTS"] = True
+    app.config["SECURITY_TOKEN_AUTHENTICATION_KEY"] = None
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/app.py:92]: Add a comment about what this means

Comment (ilija-lazoroski) [reply monkey/monkey_island/cc/app.py:94]: Done!

File: monkey/monkey_island/cc/app.py [hunk 2]
```diff
@@ -109,12 +113,31 @@ class CustomConfirmRegisterForm(ConfirmRegisterForm):
             "Email", default="dummy@dummy.com", validators=[validate_no_user_exists_already]
         )
 
+    from flask_login import user_loaded_from_request
+
+    @user_loaded_from_request.connect
+    def user_loaded_from_request(self, user=None):
+        g.login_via_request = True
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/app.py:120]: What does this mean? What is `g`?

File: monkey/monkey_island/cc/app.py [hunk 3]
```diff
@@ -109,12 +113,31 @@ class CustomConfirmRegisterForm(ConfirmRegisterForm):
...
+    class CustomSessionInterface(SecureCookieSessionInterface):
+        """Prevent creating session from API requests."""
+
+        def should_set_cookie(self, *args, **kwargs):
+            return False
+
+        def save_session(self, *args, **kwargs):
+            if g.get("login_via_request"):
+                return
+            return super(CustomSessionInterface, self).save_session(*args, **kwargs)
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/app.py:131]: Can we put this all into a method and call it like `disable_session_cookies(app)`? Or `diable_sessions`?

File: monkey/monkey_island/cc/app.py [hunk 4]
```diff
@@ -88,6 +89,9 @@ def setup_authentication(app, data_dir):
     # Ignore CSRF, because it's irrelevant for javascript applications
     app.config["WTF_CSRF_CHECK_DEFAULT"] = False
     app.config["SECURITY_CSRF_IGNORE_UNAUTH_ENDPOINTS"] = True
+    # In Token Authentication we can pass the token both in query parametar and header
+    # This disables the query parameter token
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/app.py:93]:
```suggestion
    # Forbid sending authentication token in URL parameters
```

File: monkey/monkey_island/cc/app.py (hunk 2 above)
Comment (ilija-lazoroski) [reply monkey/monkey_island/cc/app.py:120]: https://flask.palletsprojects.com/en/2.2.x/api/#flask.g
image
We are practically saying that we are going to login in from request and in that case we don't want to save a session.

File: vulture_allowlist.py [hunk 5]
```diff
@@ -102,6 +102,9 @@
 app.url_map.strict_slashes
 api.representations
 hub.exception_stream
+app.login_via_request
+app.should_set_cookie
```
Comment (VakarisZ) [inline vulture_allowlist.py:106]: Shouldn't this be `CustomSessionInterface.should_set_cookie`? What does the `app` have to do with it?

**Review by VakarisZ — APPROVED** [review]

File: monkey/monkey_island/cc/app.py (hunk 2 above)
Comment (VakarisZ) [reply monkey/monkey_island/cc/app.py:120]: Ahh, so the `g` is flasks' global context

File: vulture_allowlist.py (hunk 5 above)
Comment (ilija-lazoroski) [reply vulture_allowlist.py:106]: The vulture was complaining about the app.

File: monkey/monkey_island/cc/app.py (hunk 3 above)
Comment (ilija-lazoroski) [reply monkey/monkey_island/cc/app.py]: The class yep, the user_loaded_from_request nope.

File: monkey/monkey_island/cc/app.py [hunk 6]
```diff
@@ -114,6 +117,13 @@ class CustomConfirmRegisterForm(ConfirmRegisterForm):
         user_datastore,
         confirm_register_form=CustomConfirmRegisterForm,
     )
+    from flask_login import user_loaded_from_request
+
+    @user_loaded_from_request.connect
+    def user_loaded_from_request(self, user=None):
+        g.login_via_request = True
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/app.py:124]: this also belongs in `disable_session_cookies`?

File: monkey/monkey_island/cc/app.py [hunk 7]
```diff
@@ -4,7 +4,8 @@
 from typing import Iterable, Set, Type
 
 import flask_restful
-from flask import Flask, Response, send_from_directory
+from flask import Flask, Response, g, send_from_directory
```
Comment (shreyamalviya) [inline monkey/monkey_island/cc/app.py]: What's `g`?

**Review by shreyamalviya — APPROVED** [review]

File: monkey/monkey_island/cc/app.py (hunk 7 above)
Comment (shreyamalviya) [reply monkey/monkey_island/cc/app.py]: Ignore

**Review by VakarisZ — APPROVED** [review]

**Pull request merged**